src/NuGet.Core/NuGet.Protocol/Utility/LocalFolderUtility.cs

```diff
@@ -1170,13 +1170,7 @@ public static void GenerateNupkgMetadataFile(string nupkgPath, string installPat
                         using (var packageReader = new PackageArchiveReader(stream))
                         {
                             // get hash of unsigned content of signed package
-                            var packageHash = packageReader.GetContentHashForSignedPackage(CancellationToken.None);
-
-                            // if null, then it's unsigned package so just read the existing sha512 file
-                            if (string.IsNullOrEmpty(packageHash))
-                            {
-                                packageHash = File.ReadAllText(hashPath);
-                            }
+                            var packageHash = packageReader.GetContentHash(CancellationToken.None, hashPath);
```

Comment:
Ok, now I see it's used in 2 different places. I still think we should avoid the disk hit.